shell/server/core.js

```diff
@@ -142,59 +155,11 @@ class NotificationHandle {
       }
     });
   }
-
-  save(params) {
-    return saveFrontendRef({ notificationHandle: this.notificationId }, params.sealFor);
-  }
 }
 
 class PersistentUiViewImpl {
-  constructor(token, sturdyRef, membraneRequirements) {
-    this._token = token;
-    this._sturdyRef = sturdyRef;
-    // We need this token so we can later pass it to SandstormPermissions.createNewApiToken, which
-    // requires the sturdyRef of the token it's producing a child for, not just the hashed token id.
-
-    this._membraneRequirements = membraneRequirements || [];
-    // TODO(someday): implement SystemPersistent.addRequirements() in a way that collects
-    // requirements, and then saves them appropriately in save().
-  }
-
-  save(params) {
-    // Create a new child apiToken for the user whose session attempted to save this capability.
-    // The token should specify this.token as the parent token, and needs to gain an additional membrane
-    // requirement (TODO):
-    // * if params.sealFor is a user, then the user must still have access to the grain that offered
-    //   this cap.  Today, this is handled by offer() in SessionContextImpl.
-    const res = inMeteor(() => {
-      let owner = _.extend({}, params.sealFor);
-      const grainId = this._token.grainId;
-
-      if (owner.user) {
-        // Copy the owner's save label for this capability, if available, to be the new title.
-        if (this._token && this._token.owner && this._token.owner.grain &&
-            this._token.owner.grain.saveLabel && this._token.owner.grain.saveLabel.defaultText) {
-          owner.user.title = this._token.owner.grain.saveLabel.defaultText;
-        }
-
-        // createNewApiToken() expects only these fields if owner is a user, and will add the rest itself.
-        owner = { user: {
-          identityId: owner.user.identityId,
-          title: owner.user.title,
-        }, };
-      }
-
-      const ret = SandstormPermissions.createNewApiToken(
-          globalDb,
-          { rawParentToken: this._sturdyRef.toString() },
-          this._token.grainId,
-          this._token.petname,
-          this._token.roleAssignment,
-          owner);
-
-      return { sturdyRef: new Buffer(ret.token) };
-    });
-    return res;
+  constructor(persistentMethods) {
+    _.extend(this, persistentMethods);
```

Comment:
The problem is that the implementation of `save()` is likely to change in the future, and also we're likely to want to implement `addRequirements()` as well.

Another strategy, I suppose, would be to have all these classes inherit a common base class and then pass around an encapsulated version of the state it needs.